src/Tokenizers/PHP.php

```diff
@@ -870,6 +880,96 @@ protected function tokenize($string)
                 continue;
             }//end if
 
+            /*
+                Enum keyword for PHP < 8.1
+            */
+
+            if ($tokenIsArray === true
+                && $token[0] === T_STRING
+                && strtolower($token[1]) === 'enum'
+            ) {
+                // Get the next non-empty token.
+                for ($i = ($stackPtr + 1); $i < $numTokens; $i++) {
+                    if (is_array($tokens[$i]) === false
+                        || isset(Util\Tokens::$emptyTokens[$tokens[$i][0]]) === false
+                    ) {
+                        break;
+                    }
+                }
+
+                if (isset($tokens[$i]) === true
+                    && is_array($tokens[$i]) === true
+                    && $tokens[$i][0] === T_STRING
+                ) {
+                    // Modify $tokens directly so we can use it later when converting enum "case".
+                    $tokens[$stackPtr][0] = T_ENUM;
+
+                    $newToken            = [];
+                    $newToken['code']    = T_ENUM;
+                    $newToken['type']    = 'T_ENUM';
+                    $newToken['content'] = $token[1];
+                    $finalTokens[$newStackPtr] = $newToken;
+
+                    if (PHP_CODESNIFFER_VERBOSITY > 1) {
+                        echo "\t\t* token $stackPtr changed from T_STRING to T_ENUM".PHP_EOL;
+                    }
+
+                    $newStackPtr++;
+                    continue;
+                }
+            }//end if
+
+            /*
+                Convert enum "case" to T_ENUM_CASE
+            */
+
+            if ($tokenIsArray === true
+                && $token[0] === T_CASE
+            ) {
```

Comment:
Should this have a similar check as with the `T_DEFAULT` retokenization to prevent definitely _not_ case keywords from being retokenized ?

```php
function case() {}
```

See:
https://github.com/squizlabs/PHP_CodeSniffer/blob/5fb9b643da9e0420f979b0f87fdac4667225d621/src/Tokenizers/PHP.php#L1425
https://github.com/squizlabs/PHP_CodeSniffer/blob/5fb9b643da9e0420f979b0f87fdac4667225d621/src/Tokenizers/PHP.php#L1466-L1479

Reply:
https://github.com/squizlabs/PHP_CodeSniffer/pull/3484